File: be/SpringBoot/src/main/java/com/common/dto/request/RefreshTokenRequest.java
package com.common.dto.request;

import jakarta.validation.constraints.NotBlank;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

/**
 * DTO yêu cầu cấp lại Access Token từ Refresh Token.
 *
 * @created_at 2026-07-31
 * @author txhoan
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class RefreshTokenRequest {

    @NotBlank(message = "{auth.refresh_token_required}")
    String refreshToken;
}

File: be/.Net/Swe.BL/DTO/Request/RefreshTokenRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Swe.BL.DTO.Request;

/// <summary>
/// DTO yêu cầu làm mới Token.
/// </summary>
/// <created_at>2026-08-03</created_at>
/// <author>txhoan</author>
public class RefreshTokenRequest
{
    [Required(ErrorMessageResourceType = typeof(Swe.Common.Resources.ResourcesVN), ErrorMessageResourceName = "RefreshTokenRequired")]
    public string RefreshToken { get; set; } = string.Empty;
}
